### PR TITLE
Make category CRUD for Admin

### PR DESCRIPTION
**Pull request by Tomoki108**
***Commands to run***
-none

***Preconditions***
-Go to url(hostname/admin/category).
-Check CRUD functions using the website.

***Expected outputs***
-Categories, Add category and Edit category  pages are displayed properly. The design is based on wireframes.
https://docs.google.com/presentation/d/1a0mVYW0HPXVJn5pCqQp9an6AA6Kuq515HaVuDW9b030/edit#slide=id.g3e604c4bec_1_5

-All of CRUD functions should work. (Category page takes a roll as Read function.)

***asana***
-https://app.asana.com/0/984197702640917/984490302593925

***note***
-There is no authentication function yet. Anyone can manage category for now. Later I'll make it.

### CODE REVIEW

File: app/Category.php [hunk 1]
```diff
@@ -10,11 +10,25 @@ class Category extends Model
...
     {
         $this->title = $request->title;
         $this->description = $request->description;
         $this->save();
         return $this;
     }
+
+    public function updateCategory($request, $category)
+    {
+        $category->title = $request->title;
```
Comment (jeremiahC) [inline app/Category.php]: no need to use category just use this and no need to pass  category object

File: app/Category.php [hunk 2]
```diff
@@ -10,11 +10,25 @@ class Category extends Model
     
     public $timestamps = false;
 
+    public function index($number)
+    {
+        $categories = $this->orderBy('id', 'DESC')->paginate($number);
+        return $categories;
```
Comment (jeremiahC) [inline app/Category.php]: it's ok no need to put it in variable, it's ok like this `return $this->orderBy('id', 'DESC')->paginate($number);`

File: app/Category.php [hunk 3]
```diff
@@ -10,11 +10,25 @@ class Category extends Model
     
     public $timestamps = false;
 
+    public function index($number)
```
Comment (jeremiahC) [inline app/Category.php]: name functions more meaning fool and readable, like `getCategories();`

File: resources/views/admin/category.blade.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
+@extends ('layouts.app')
+
+@section('title', 'Categories')
+
+@section('admin_mark', ' | Admin')
+
+@section('content')
+<style>
+</style>
```
Comment (jeremiahC) [inline resources/views/admin/category.blade.php]: please delete this style tag if no use

File: resources/views/admin/category.blade.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    <section>
+        <table class="table table-hover">
+          <thead>
+              <tr><th>Title</th><th>Description</th><th>Action</th></tr>
+          </thead>
+
+          <tbody>
+              @foreach ($categories as $category)
+                  <?php
```
Comment (jeremiahC) [inline resources/views/admin/category.blade.php]: you don't need to create a variable, creating variable consumes memory of the computer, you can assign it directly `href=""`

File: resources/views/flashMessage.blade.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+<div class="alert alert-warning alert-block">
+	<button type="button" class="close" data-dismiss="alert">×</button>	
+	<strong>{{ $message }}</strong>
+</div>
+@endif
+
+
+@if ($message = Session::get('info'))
+<div class="alert alert-info alert-block">
+	<button type="button" class="close" data-dismiss="alert">×</button>
```
Comment (jeremiahC) [inline resources/views/flashMessage.blade.php]: fix indentation

File: resources/views/layouts/app.blade.php [hunk 7]
```diff
@@ -11,7 +11,7 @@
 
     <!-- Scripts -->
     <script src="{{ asset('js/app.js') }}" defer></script>
-
+
```
Comment (jeremiahC) [inline resources/views/layouts/app.blade.php]: remove trailing spaces

File: resources/views/admin/addCategory.blade.php [hunk 8]
```diff
@@ -12,7 +12,6 @@
             <h3 class="mb-3">Add category</h3>
             <h5>Title</h5>
             <input id="title" type="text" class="form-control{{ $errors->has('title') ? ' is-invalid' : '' }} mb-4 col-12" name="title" value="{{ old('title') }}" required>
-
             <h5>Description</h5>
             <textarea id="description" class="form-control{{ $errors->has('description') ? ' is-invalid' : '' }} mb-4 col-12" name="description" rows="8" required>{{ old('description') }}</textarea>
             <button type="submit" class="btn btn-primary w-100">submit</button>
```
Comment (justinPCM) [inline resources/views/admin/addCategory.blade.php]: Change the text "submit" to "Submit". To follow the given wireframe.
Note: Only the button text, not the type.

File: routes/web.php [hunk 9]
```diff
@@ -19,18 +19,19 @@
...
     return view('admin.addCategory');
 });
-
-//temporary routing for admin to add category 
-Route::post('/create_category', 'CategoryController@create');
+Route::get('/admin/category', 'CategoryController@index');
+Route::post('/admin/category/create', 'CategoryController@create');
+Route::get('/admin/category/edit/{category}', 'CategoryController@edit');
+Route::post('/admin/category/update/{category}', 'CategoryController@update');
+Route::get('/admin/category/{category}', 'CategoryController@delete');
```
Comment (justinPCM) [inline routes/web.php]: Let's change "Route::get('/admin/category/{category}', 'CategoryController@delete');" to "Route::get('/admin/category/delete/{category}', 'CategoryController@delete');" to stay uniformed with your implementation.

File: resources/views/admin/editCategory.blade.php [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    <form method="post" action="{{ action('CategoryController@update', $category) }}">
+        {{csrf_field()}}
+        <h3 class="mb-5">Edit category</h3>
+        <h5>Title</h5>
+        <input id="title" type="text" class="form-control{{ $errors->has('title') ? ' is-invalid' : '' }} mb-4 col-12" name="title" value="{{ old('title', $category->title) }}"  required>
+
+        <h5>Description</h5>
+        <textarea id="description" class="form-control{{ $errors->has('description') ? ' is-invalid' : '' }} mb-4 col-12" name="description" rows="8" required>{{ old('description', $category->description) }}</textarea>
+        
+        <button type="submit" class="btn btn-primary w-100">submit</button>
```
Comment (justinPCM) [inline resources/views/admin/editCategory.blade.php]: Please also change "submit" to "Submit" for this one.

File: resources/views/admin/addCategory.blade.php [hunk 11]
```diff
@@ -12,10 +12,9 @@
             <h3 class="mb-3">Add category</h3>
             <h5>Title</h5>
             <input id="title" type="text" class="form-control{{ $errors->has('title') ? ' is-invalid' : '' }} mb-4 col-12" name="title" value="{{ old('title') }}" required>
```
Comment (justinPCM) [inline resources/views/admin/addCategory.blade.php]: In the database, the maximum number of characters you can input for the title is only 191. You should add a "maxlength" attribute on your input element to restrict the number of characters entered. Also do this for "edit category". Any input exceeding 191 characters for title would result to an error in you current implementation.

**Review by justinPCM — APPROVED** [review]

**Pull request merged**